index.js

```diff
@@ -51,7 +51,9 @@ const createAdapter = handler => config => new Promise((resolve, reject) => {
   const server = handler instanceof Server ? handler : createServer(handler)
   const listening = server.listening
 
-  server.on('error', reject);
+  if (server.eventNames().includes('error')) {
```

Comment:
Does it setup error handler only if there is already one?

Reply:
No, because the server listens for errors using `on`, not `once`. You can try emitting the error many times.